server/src/main/java/com/cloud/user/AccountManagerImpl.java

```diff
@@ -358,10 +391,46 @@ public UserVO getSystemUser() {
 
     @Override
     public boolean start() {
+        if (apiNameList == null) {
+            long startTime = System.nanoTime();
+            apiNameList = new ArrayList<String>();
+            Set<Class<?>> cmdClasses = new LinkedHashSet<Class<?>>();
+            for (PluggableService service : services) {
```

Comment:
Should we only get api name list from the configured pluggable service (the dynamic or static manager?)

Reply:
this implementation is cleverly stolen. I'll have a look.

Reply:
No, we need the full API list by name, the plugable service hold class references, which is overkill for us at this moment.

Reply:
the role service can be dis-/enabled dynamically. we need a conclusive list of all API. It needs to be done independent of which checker is active.